RecoParticleFlow/PFClusterProducer/interface/PFEcalBarrelRecHitCreator.h

```diff
@@ -0,0 +1,254 @@
+#ifndef RecoParticleFlow_PFClusterProducer_PFEcalBarrelRecHitCreator_h
+#define RecoParticleFlow_PFClusterProducer_PFEcalBarrelRecHitCreator_h
+
+#include "RecoParticleFlow/PFClusterProducer/interface/PFRecHitCreatorBase.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
+
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "Geometry/CaloGeometry/interface/TruncatedPyramid.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+#include "Geometry/EcalAlgo/interface/EcalEndcapGeometry.h"
+#include "Geometry/EcalAlgo/interface/EcalBarrelGeometry.h"
+#include "Geometry/CaloTopology/interface/EcalEndcapTopology.h"
+#include "Geometry/CaloTopology/interface/EcalBarrelTopology.h"
+#include "Geometry/CaloTopology/interface/EcalPreshowerTopology.h"
+#include "Geometry/CaloTopology/interface/EcalTrigTowerConstituentsMap.h"
+#include "RecoCaloTools/Navigation/interface/CaloNavigator.h"
+
+template <typename Geometry,PFLayer::Layer Layer,int Detector>
+  class PFEcalBarrelRecHitCreator :  public  PFRecHitCreatorBase {
+
+ public:  
+  PFEcalBarrelRecHitCreator(const edm::ParameterSet& iConfig,edm::ConsumesCollector& iC):
+    PFRecHitCreatorBase(iConfig,iC)
+    {
+      recHitToken_ = iC.consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("src"));
+      srFlagToken_ = iC.consumes<EBSrFlagCollection>(iConfig.getParameter<edm::InputTag>("srFlags"));
+      SREtaSize_ = iConfig.getUntrackedParameter<int> ("SREtaSize",1);
+      SRPhiSize_ = iConfig.getUntrackedParameter<int> ("SRPhiSize",1);
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.